galaxy: fix selects and more

bernt-matthias opened the pull request:
- check only use compatible display attribute for selects, see also https://github.com/galaxyproject/galaxy/issues/11797

TODO: 
- [ ] test for `output-prefix` https://github.com/WorkflowConversion/CTDConverter/issues/61

bernt-matthias committed:
galaxy: fix selects

https://github.com/galaxyproject/galaxy/issues/11797

bernt-matthias committed:
add possibility to batch mode for multiple=true data parameters

bernt-matthias committed:
fixup: add possibility to batch mode for multiple=true data parameters

bernt-matthias committed:
fix parameter path for batch mode parameter

and fixup for out type parameter in tests

bernt-matthias committed:
also add batch conditional in tests

and use delta_frac fot test outputs

bernt-matthias committed:
also treat optional parameters with default as optional

bernt-matthias committed:
make output file comparison configurable

eg `--test-condition "compare=sim_size" "delta_frac=0.05"`
which is the default

bernt-matthias committed:
insert test macros instead of import

jpfeuffer commented:
Looks ok. Let me know when you want to merge. Btw, do you think it makes sense to give you write/admin rights here?

bernt-matthias commented:
> Let me know when you want to merge.

I guess when I'm happy with the 2.8 galaxy wrappers over here https://github.com/galaxyproteomics/tools-galaxyp/pull/653 and tests run successful again...

I will send you a ping.

> Btw, do you think it makes sense to give you write/admin rights here?

If you like.

bernt-matthias commented:
@jpfeuffer any idea why the travis tests are not running here?

bernt-matthias committed:
rename GALAXY_VERSION to VERSION_SUFFIX

bernt-matthias committed:
make advanced options a select

instead of conditional

bernt-matthias committed:
ignore duplicate options for selects

bernt-matthias committed:
fix test type parameters

bernt-matthias committed:
only use 1st line of description

for Galaxy tool description

see https://github.com/OpenMS/OpenMS/issues/6568

bernt-matthias commented:
Any idea why CI tests are not starting here?

bernt-matthias committed:
only flatten lists and input needs no default

bernt-matthias committed:
only flatten lists

and make code a bit more readable (hopefully)

bernt-matthias committed:
use logging from stdlib

bernt-matthias committed:
remove some unused cli arguments

for the Galaxy converter

bernt-matthias committed:
add github actions test

bernt-matthias committed:
force installation of requirements.txt for lint

and remove copy paste remnants

bernt-matthias committed:
skip flake E501 and add requirements

bernt-matthias committed:
also run on push

bernt-matthias committed:
disable black, fill requirements.txt

bernt-matthias committed:
temp disable diff_pydocstyle_report

bernt-matthias committed:
galaxy: do not set name and argument

bernt-matthias committed:
galaxy: add a mock count attribute to test/ouput_collection elements

count is now required for planemo lint to pass

bernt-matthias commented:
Hey @jpfeuffer lets get tests running first https://github.com/WorkflowConversion/CTDConverter/pull/64

bernt-matthias committed:
itry to run only on pull_request

bernt-matthias committed:
Merge pull request #4 from bernt-matthias/topic/gha-tool-linting-fixes

Topic/gha tool linting fixes

bernt-matthias committed:
post rebase cleanup

bernt-matthias committed:
fixup for batch mode

tests

bernt-matthias committed:
fixup: for batch mode

test data generation needs to remove whens and set the select parameter

bernt-matthias committed:
fix test generation

- remove XML declaration from output
- indent XML

bernt-matthias committed:
some fixes

bernt-matthias committed:
copy input files instead of symlinking

bernt-matthias committed:
also do not make output_prefix parameters advanced

bernt-matthias committed:
better support for out_prefix and do not use checkboxes/radio

bernt-matthias committed:
ommit optional if default

bernt-matthias commented:
@jpfeuffer https://github.com/WorkflowConversion/CTDConverter/pull/63/commits/767f14814a4037381806adfcd898d1bed01d5c89 was used to generate the Galaxy tools for 3.1 (which have now been released).

bernt-matthias committed:
imporve prefix mapping

some OpenMS tools have  parameters with the same name (different
sections). This creates problems in Galaxy.

workaround is to hardcode a unique prefix.

Before _ was used which now creates problems with the tool
linting (parameters are not supposed to start with _).

Hence we now use prefix of the form DEDUP_[0-9]+_DEDUP_
